### DISCUSSION TITLE
is is the expected behavior? actor.stop() call complete() subscription

### DISCUSSION THREAD

**Discussion by christophe-g** [statelyai/xstate#4608] [Q&A]
It seems that stopping an actor with a `complete` subscription will call the complete. 

```js
const subscription = actor.subscribe({
  complete() {
    console.info('I AM CALLED')
  },
});

actor.stop()
```

My (probably wrong) expectation is that complete is only called when a final state is reached, not when an actor is stopped. If this is the expected behavior, is it possible to stress that in the docs ? 

I cannot see test for this behavior.

**Comment by Andarist** [comment]
This is a somewhat disputable behavior - but overall, I think that I agree. `complete` should only be called when the actor reaches a done snapshot.
